contrib/coredns/policy/policy.go

```diff
@@ -150,7 +196,54 @@ func (p *PolicyMiddleware) ServeDNS(ctx context.Context, w dns.ResponseWriter, r
 	}
 
 	if response.Permit {
-		return middleware.NextOrFailure(p.Name(), p.Next, ctx, w, r)
+
+		lr := new(LocalResponseWriter)
```

Comment:
Use NewLocalResponseWriter instead and then you can get rid of the three lines that follow this.